### PR TITLE
coprocessor: reduce the slow log count by changing the req_lifetime to total_process_time

### PR DESCRIPTION
**Pull request by L3T**
### What problem does this PR solve?

#10841 *:Logger thread can be bottleneck and slow down the whole TiKV instance 

Problem Summary:

### What is changed and how it works?

Proposal: #10841

What's Changed:

### Related changes

- Need to cherry-pick to the release branch
release-3.0
release-3.1
release-4.0
release-5.0
release-5.1
release-5.2

### Check List 

Tests 

- Manual test (add detailed scripts or steps below)
- No code

Side effects

- Slow log will only consider time spent on processing the request. 
- Drop log instead of blocking threads when slogger thread is overloaded and queue is filled up. 

### Release note 

```release-note
- TiKV coprocessor slow log will only consider time spent on processing the request. 
- Drop log instead of blocking threads when slogger thread is overloaded and queue is filled up. 
```

### CODE REVIEW

**Review by sunxiaoguang — APPROVED** [review]
LGTM

**Review by skyzh — APPROVED** [review]
LGTM. Are you sure to cherry-pick this commit to so many releases?

**Comment by skyzh** [comment]
Ping us when you are ready to merge. Feel free to add or remove releases you want to cherry-pick.

**Comment by L3T** [comment]
I`m Ready to Merge.

File: components/tikv_util/src/logger/mod.rs [hunk 1]
```diff
@@ -31,7 +31,7 @@ pub const DATETIME_ROTATE_SUFFIX: &str = "%Y-%m-%d-%H:%M:%S%.f";
 const SLOG_CHANNEL_SIZE: usize = 10240;
 // Default is DropAndReport.
 // It is not desirable to have dropped logs in our use case.
-const SLOG_CHANNEL_OVERFLOW_STRATEGY: OverflowStrategy = OverflowStrategy::Block;
+const SLOG_CHANNEL_OVERFLOW_STRATEGY: OverflowStrategy = OverflowStrategy::Drop;
```
Comment (zhongzc) [inline components/tikv_util/src/logger/mod.rs:34]: Comments need to be updated.

Comment (zhongzc) [reply components/tikv_util/src/logger/mod.rs:34]: Would `DropAndReport` be better?

Comment (sunxiaoguang) [reply components/tikv_util/src/logger/mod.rs:34]: @L3T please take a look at comment. You can file a new PR to address comment. And make sure all the cherrypick PRs have comments updated, thanks.

Comment (sunxiaoguang) [reply components/tikv_util/src/logger/mod.rs:34]: > Would `DropAndReport` be better?

Good question, we kind of discussed this strategy as well. The agreement was overflowing message isn't much useful to both users and developers therefore `Drop` is an appropriate choice. However it is definitely open to discussion before our long term solution is implemented.

Comment (zhongzc) [reply components/tikv_util/src/logger/mod.rs:34]: Thanks for your reply.

A drop report log can inform us the logger is busy in a cheap way. Although we are working on letting this information be shown on grafana, I feel better with an error can bring many different forms of warnings.

Comment (sunxiaoguang) [reply components/tikv_util/src/logger/mod.rs:34]: DropAndReport inform us by a line of log message which is for human being only. We can neither know anything about it from metrics nor the context of dropped log was about. And that's why we thought it was not useful at the moment and used Drop instead.

**Pull request merged**